src/sdkV2.ts

```diff
@@ -0,0 +1,111 @@
+import { ethers } from 'ethers';
+import { EvmAddress, TxOverrides } from './types';
+import { GasStation } from './GasStation';
+
+import Erc20Abi from './abi/ERC20.json';
+import IDCSEntryAbi from './abiV2/IDCSEntry.json';
+
+export default class CegaEvmSDKV2 {
+  private _provider: ethers.providers.Provider;
+
+  private _signer: ethers.Signer | undefined;
+
+  private _gasStation: GasStation;
+
+  private _cegaEntryAddress: EvmAddress;
+
+  constructor(
+    cegaEntryAddress: EvmAddress,
+    gasStation: GasStation,
+    provider: ethers.providers.Provider,
+    signer: ethers.Signer | undefined = undefined,
+  ) {
+    this._provider = provider;
+    this._signer = signer;
+    this._gasStation = gasStation;
+    this._cegaEntryAddress = cegaEntryAddress;
+  }
+
+  setProvider(provider: ethers.providers.Provider) {
+    this._provider = provider;
+  }
+
+  setSigner(signer: ethers.Signer) {
+    this._signer = signer;
+  }
+
+  setCegaEntryAddress(cegaEntryAddress: EvmAddress) {
+    this._cegaEntryAddress = cegaEntryAddress;
+  }
+
+  loadCegaEntry(): ethers.Contract {
+    return new ethers.Contract(
+      this._cegaEntryAddress,
+      IDCSEntryAbi.abi,
+      this._signer || this._provider,
+    );
+  }
+
+  async getProductDcs(productId: ethers.BigNumberish) {
+    const cegaEntry = this.loadCegaEntry();
+    return cegaEntry.getDCSProduct(productId);
+  }
+
+  async getLatestProductIdDcs() {
```

Comment:
QQ - what's this for?

Reply:
it's just a function available on the SC. I added it as a getter.

can remove it